ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -58,56 +59,107 @@ func NewReadWriter() (*YAMLReadWriter, error) {
+	if err = yaml.Unmarshal(dat, &config); err != nil {
+		return errors.Wrap(err, "Failed to parse yaml file: "+path)
 	}
 
-	// convert yaml to a map (replaces IsKeyPresent functionality)
-	if err = yaml.Unmarshal(dat, &configMap); err != nil {
-		return errors.Wrapf(err, "Error parsing yaml file %s", yamlPath)
+	// get the correct cluster
+	var chosenCluster string
+	if clusterConfigKey == "" {
+		chosenCluster = config.Default
+	} else {
+		chosenCluster = clusterConfigKey
 	}
 
+	cluster := config.Clusters[chosenCluster]
+	// Get the info out of the cluster
+	cliConfig.Region = cluster.Region
+	cliConfig.Cluster = cluster.Cluster
+	cliConfig.ComposeServiceNamePrefix = cluster.ComposeServiceNamePrefix
 	return nil
+
 }
 
-// GetConfig gets the ecs-cli config object from the config file.
-// map contains the keys that are present in the config file (maps string field name to string field value)
-// map is type map[interface{}]interface{} to ensure fowards compatibility with changes that will
-// cause certain keys to be mapped to maps of keys
-func (rdwr *YAMLReadWriter) GetConfig() (*CLIConfig, map[interface{}]interface{}, error) {
-	to := new(CLIConfig)
-	configMap := make(map[interface{}]interface{})
-	// read the raw bytes of the config file
-	path := configPath(rdwr.destination)
-
-	if errYAML := readYAML(path, configMap, to); errYAML == nil {
-		// File is YAML
-		return to, configMap, nil
-	} else if to, configMap, errINI := readINI(rdwr.destination); errINI == nil {
-		// File is INI
-		return to, configMap, nil
-	} else if errINI != nil {
-		// Return the ini error
-		// This will return a parsing error for ini (format error)
-		return nil, nil, errINI
+// readProfileConfig does all the work to read and parse the yaml cluster config
+func readProfileConfig(path string, profileConfigKey string, cliConfig *CLIConfig) error {
+	// read cluster file
+	dat, err := ioutil.ReadFile(path)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read config file: "+path)
+	}
+
+	config := ProfileConfig{Profiles: make(map[string]Profile)}
+	if err = yaml.Unmarshal(dat, &config); err != nil {
+		return errors.Wrap(err, "Failed to parse yaml file: "+path)
+	}
+
+	// get the correct profile
+	var chosenProfile string
+	if profileConfigKey == "" {
+		chosenProfile = config.Default
 	} else {
-		// If yaml through an error, but ini didn't throw an error- return the yaml error
-		return nil, nil, errYAML
+		chosenProfile = profileConfigKey
 	}
+	profile := config.Profiles[chosenProfile]
+
+	// Get the info out of the cluster
+	cliConfig.AWSSecretKey = profile.AWSSecretKey
+	cliConfig.AWSAccessKey = profile.AWSAccessKey
+
+	return nil
+
+}
+
+// GetConfigs gets the ecs-cli config object from the config file(s).
+// This function either reads the old single configuration file
+// Or if the new files are present, it reads from them instead
+func (rdwr *YAMLReadWriter) GetConfigs(clusterConfig string, profileConfig string) (*CLIConfig, error) {
```

Comment:
Since we won't have multiple get config functions, What do you think of renaming this to "Get"?